### PR TITLE
feat(error): revamp `make_canonical` error

### PR DESCRIPTION
**Pull request by tcoratger**
Should solve #2752.

I added a dedicated `CanonicalError` enum and embedded that inside `reth_interfaces::error::Error`. I don't know whether to also change the declaration of the function `pub fn make_canonical(&mut self, block_hash: &BlockHash) -> Result<CanonicalOutcome, Error>` to `pub fn make_canonical(&mut self, block_hash: &BlockHash) -> Result<CanonicalOutcome, CanonicalError>` but I think that would induce many other changes which are not desired, am I wrong?

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
I like this.

pending @Rjected 

> but I think that would induce many other changes which are not desired, am I wrong?

agree

**Review by Rjected — APPROVED** [review]
This looks good to me pending tests passing / checks updated

**Comment by Rjected** [comment]
Looks like the tests still need to be fixed, @tcoratger do you need any help on this?

**Comment by tcoratger** [comment]
@Rjected Yes sure, I thought my last commit solved the problem but apparently not. I feel like the CI tests stop because many of them are too slow. But I don't understand why because nothing fundamental changes here? Maybe I missed something.

**Comment by tcoratger** [comment]
@mattsse Do you know why the tests fail here?

File: crates/consensus/beacon/src/engine/mod.rs [hunk 1]
```diff
@@ -907,10 +907,8 @@ where
 
         #[allow(clippy::single_match)]
         match &error {
-            Error::Execution(
-                error @ BlockExecutionError::Validation(BlockValidationError::BlockPreMerge {
-                    ..
-                }),
+            Error::Canonical(
```
Comment (tcoratger) [inline crates/consensus/beacon/src/engine/mod.rs:910]: @mattsse I think I spotted the bug, this one had to be modified too. Now it should be good

**Review by mattsse — APPROVED** [review]
I like this, this is a lot cleaner

**Pull request merged**